File: exercise1/src/main/java/exercise1/logic/exceptions/CustomerAlreadyExistExecption.java
package exercise1.logic.exceptions;

import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.http.HttpStatus;

@ResponseStatus(code = HttpStatus.INTERNAL_SERVER_ERROR) // ERROR 500 CODE
public class CustomerAlreadyExistExecption extends RuntimeException {
	private static final long serialVersionUID = -6115904545317055186L;

	public CustomerAlreadyExistExecption() {
	}

	public CustomerAlreadyExistExecption(String message) {
		super(message);
	}

	public CustomerAlreadyExistExecption(Throwable cause) {
		super(cause);
	}

	public CustomerAlreadyExistExecption(String message, Throwable cause) {
		super(message, cause);
	}

}

File: Demo - Kotlin/src/main/kotlin/demo/CustomerAlreadyExistExecption.kt
package demo

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus


@ResponseStatus(code = HttpStatus.INTERNAL_SERVER_ERROR) // ERROR 500 CODE
class CustomerAlreadyExistExecption : RuntimeException {
    constructor() {}
    constructor(message: String?) : super(message) {}
    constructor(cause: Throwable?) : super(cause) {}
    constructor(message: String?, cause: Throwable?) : super(message, cause) {}

    companion object {
        private const val serialVersionUID = -6115904545317055186L
    }
}
